src/pineko/cli/check.py

```diff
@@ -30,7 +30,10 @@ def sub_compatibility(grid_path, operator_path, xif):
 
     """
     pineappl_grid = pineappl.grid.Grid.read(grid_path)
-    operators = eko.output.legacy.load_tar(operator_path)
+    if check.check_eko_is_legacy(operator_path):
+        operators = eko.output.legacy.load_tar(operator_path)
+    else:
+        operators = eko.output.EKO.load(operator_path)
```

Comment:
Pattern repeated third time, the function is really the best choice at this point :)